src/PrestaShopBundle/Controller/Admin/Sell/Catalog/Product/ProductController.php

```diff
@@ -1877,4 +1879,50 @@ private function getShopIdFromShopContext(): ?int
 
         return !empty($shopId) ? (int) $shopId : null;
     }
+
+    /**
+     * Displays a category tree.
+     *
+     * @param Request $request
+     *
+     * @return Response
+     */
+    #[AdminSecurity("is_granted('create', request.get('_legacy_controller')) || is_granted('update', request.get('_legacy_controller')) || is_granted('read', request.get('_legacy_controller'))")]
+    public function categoryTreeAction(
```

Comment:
```suggestion
    public function legacyCategoryTreeAction(
```
Can you please rename this action, so we understand right away that it's related to legacy pages and ideally add a small comment explaining that it should be removed someday when all pages relying on this component will have been migrated